ideascube/mediacenter/tests/test_models.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+
+from .factories import DocumentFactory
+
+from os.path import basename
+
+@pytest.mark.django_db(transaction=True)
+def test_deleting_document_should_delete_file_also(mocker):
+    file_name = 'foo.img'
+    exist_mock = mocker.patch('os.path.exists')
+    exist_mock.side_effect = lambda p: basename(p) == file_name
+    remove_mock = mocker.patch('os.remove')
+    document = DocumentFactory(original=file_name)
+    document.delete()
+    assert remove_mock.call_count == 1
+
+
+@pytest.mark.django_db(transaction=True)
+def test_deleting_document_should_delete_preview_also(mocker):
+    file_name = 'preview_foo.img'
+    exist_mock = mocker.patch('os.path.exists')
+    exist_mock.side_effect = lambda p: basename(p) == file_name
+    remove_mock = mocker.patch('os.remove')
+    document = DocumentFactory(preview=file_name)
+    document.delete()
+    assert remove_mock.call_count == 1
+
+
+@pytest.mark.django_db(transaction=True)
+def test_deleting_document_should_delete_file_and_preview_also(mocker):
+    file_name = 'oo.img'
+    preview_name = 'preview_foo.img'
+    exist_mock = mocker.patch('os.path.exists')
+    exist_mock.side_effect = lambda p: basename(p) in (file_name, preview_name)
+    remove_mock = mocker.patch('os.remove')
```

Comment:
Should you really mock these two?

Mocking them prevents you from testing what happens if an `OSError` gets raised while removing the files.

And in fact, did you test it? What happens if the file to be removed doesn't exist any more, for example? (e.g the admin `rm -fr`-ed it)

Basically, given its place in the transaction (at `on_commit` time), `real_delete` just cannot fail, or else the transaction will be botched.